crypto/cmp/cmp_msg.c

```diff
@@ -994,6 +996,8 @@ X509 *ossl_cmp_certresponse_get1_certificate(EVP_PKEY *privkey,
     }
     if (crt == NULL)
         CMPerr(0, CMP_R_CERTIFICATE_NOT_FOUND);
+    else
+        (void)x509_set0_libctx(crt, ctx->libctx, ctx->propq);
```

Comment:
In general I'd agree that it should be done as early as possible,
but in this case it's simpler to do it once at the given place 
because otherwise it would be needed not only in the `OSSL_CMP_CERTORENCCERT_ENCRYPTEDCERT` case but also in the OSSL_CMP_CERTORENCCERT_CERTIFICATE case (where `X509_dup(coec->value.certificate)` is called).

Reply:
Which means that coec->value.certificate should also already be setup straight after it is loaded also..